File: e019.js
let countdays = 365, daysOfMonth, sum = 0;
for (let yyyy = 1901; yyyy <= 2000; yyyy++) {
    for (let mm = 0; mm < 12; mm++) {
        if (mm === 1) {  // February
            if (yyyy % 4 === 0 && yyyy % 100 !== 0 || yyyy % 400 === 0) {  // Leap year
                daysOfMonth = 29;
            } else { // other year
                daysOfMonth = 28;
            }
        } else if ([8, 3, 5, 10].includes(mm)) {
            daysOfMonth = 30;
        } else {
            daysOfMonth = 31;
        }
        for (let dd = 1; dd <= daysOfMonth; dd++) {
            countdays++;
            if (countdays % 7 === 0 && dd === 1) {
                sum++;
            }
        }
    }
}
console.log(sum);

File: e019.py
"""
Problem 19 - Counting Sundays
"""

COUNTDAYS = 365
SUM = 0
for yy in range(1901, 2001):
    for mm in range(12):
        if mm == 1:
            if yy % 4 == 0 and yy % 100 != 0 or yy % 400 == 0:
                DAYS_OF_MONTH = 29
            else:
                DAYS_OF_MONTH = 28
        elif mm in [8, 3, 5, 10]:
            DAYS_OF_MONTH = 30
        else:
            DAYS_OF_MONTH = 31
        for dd in range(1, DAYS_OF_MONTH+1):
            COUNTDAYS += 1
            if COUNTDAYS % 7 == 0 and dd == 1:
                SUM += 1

print(SUM)


#===============================================
# from datetime import datetime
#
# count = 0
# for yy in range(1901, 2001):
#     for mm in range(1, 13):
#         if datetime(yy, mm, 1).weekday() == 6:
#             count += 1
# print(count)
#===============================================
